include/wx/defs.h

```diff
@@ -279,6 +279,17 @@ typedef short int WXTYPE;
     #define wxNODISCARD
 #endif
 
+/* wxWARN_UNUSED is used as an attribute to a class, stating that unused instances
+   should be warned about (in case such warnings are enabled in the first place) */
+
+#ifdef __has_attribute
+#if __has_attribute(warn_unused)
+#define wxWARN_UNUSED __attribute__((warn_unused))
+#endif
+#endif
+#ifndef wxWARN_UNUSED
+#define wxWARN_UNUSED
```

Comment:
Also, this is really nitpicky, but I'd indent this block as we do elsewhere, i.e.

```suggestion
    #if __has_attribute(warn_unused)
        #define wxWARN_UNUSED __attribute__((warn_unused))
     #endif
#endif
#ifndef wxWARN_UNUSED
    #define wxWARN_UNUSED

```